ui/lib/pki/addon/routes/roles/role/details.js

```diff
@@ -1,3 +1,11 @@
-import Route from '@ember/routing/route';
+import PkiRolesIndexRoute from '../index';
 
-export default class RolesRoleDetailsRoute extends Route {}
+export default class RolesRoleDetailsRoute extends PkiRolesIndexRoute {
```

Comment:
This route extends `roles/index` so that the role model is hydrated before the route model hook

Reply:
I'm a bit confused by this could you explain further?

Reply:
Sure! Essentially, we want this route to run `return this.pathHelp.getNewModel('pki/role', 'pki');` in the beforeModel hook so that they model is properly generated before we query it, and rather than repeat that code within this route I opted to just extend the parent route.

Reply:
Ahhh gotcha and then override the model hook to return what you need! Thanks for the explanation. Great idea!